sql: fix subqueries with prepare panic

RaduBerinde opened the pull request:
Fixing a failure I noticed in the bank benchmark: don't call splitFilter in
prepareOnly mode - subqueries don't get expanded in this case which doesn't work
with splitFilter.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4765)

<!-- Reviewable:end -->

petermattis commented:
LGTM. Is there a test case that can be added?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4765#-:-KBp5lc9AH5lZKMKl-ql:855555131)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Added test, confirmed it paniced before. BTW we should look at expanding logic_test with the capability of doing prepared queries, the framework in pgwire_test is very limited (e.g. we can't have statements in a certain order because they're in a map; all queries have to return exactly one row)

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4765#-:-KBreYvsGWzmVsFlD-lx:582219)*
<!-- Sent from Reviewable.io -->

petermattis commented:
LGTM

Yes, we should support placeholders and prepared queries in the logic test framework. See https://github.com/cockroachdb/cockroach/issues/3854.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4765#-:-KBrktnRsX8IXLlLPpHr:-824671780)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: fix subqueries with prepare panic

Fixing a failure I noticed in the bank benchmark: don't call splitFilter in
prepareOnly mode - subqueries don't get expanded in this case which doesn't work
with splitFilter.

Fixes #4765.
